app/helpers/form_helper.py

```diff
@@ -137,3 +142,12 @@ def clear_other_text_field(data, questions_for_block):
                 form_data[answer['id']] = ''
 
     return form_data
+
+
+def is_valid_form(form):
+    is_valid = form.validate()
+    if form.csrf_token.errors:
+        logger.warn('Invalid csrf token')
+        raise Forbidden
```

Comment:
Yeah, I lazily transplanted it from (here)[https://github.com/ONSdigital/eq-survey-runner/blob/master/app/views/questionnaire.py#L409]. I'll give (a custom error handler)[http://flask-wtf.readthedocs.io/en/stable/csrf.html#customize-the-error-response) a go. It should hopefully just slot straight in.

Reply:
While I'm here, technically this should probably be a Bad Args rather than Forbidden. Is there a specific reason it's a Forbidden error code currently?

Reply:
I don't think so. Looking at FlaskWTF there is a [CSRFError](http://flask-wtf.readthedocs.io/en/stable/api.html#flask_wtf.csrf.CSRFError) that generates a 400 error.